File: packages/native_crypto_android/android/src/main/kotlin/fr/pointcheval/native_crypto_android/utils/CipherAlgorithm.kt
package fr.pointcheval.native_crypto_android.utils

import fr.pointcheval.native_crypto_android.ciphers.AES
import fr.pointcheval.native_crypto_android.interfaces.Cipher

enum class CipherAlgorithm {
    aes;

    fun getCipher(): Cipher {
        return when (this) {
            aes -> AES()
        }
    }
}

File: packages/native_crypto_ios/ios/Classes/utils/CipherAlgorithm.swift
//
//  CipherAlgorithm.swift
//  native_crypto_ios
//
//  Created by Hugo Pointcheval on 25/05/2022.
//

import Foundation

enum CipherAlgorithm : String {
    case aes = "aes"
    
    var getCipher: Cipher {
        switch self {
        case .aes: return AESCipher()
        }
    }
}
